igc_lib.py

```diff
@@ -71,7 +74,148 @@ def _rawtime_float_to_hms(timef):
+            For the goal, at the moment, Turnpoints can't handle goal cones or lines, for this reason we default to goal_cylinder.
+        """
+        
+        turnpoints = []
+                
+        # Open XML document using minidom parser
+        DOMTree = xml.dom.minidom.parse(filename)
+        task = DOMTree.documentElement
+
+        # Get the taskpoints, waypoints and time gate
+        taskpoints = task.getElementsByTagName("taskpoints")[0]
+        waypoints = task.getElementsByTagName("waypoints")[0]
+        gate = task.getElementsByTagName("time-gate")[0]
+
+        tpoints = taskpoints.getElementsByTagName("point")
+
+        wpoints = waypoints.getElementsByTagName("point")
+
+        start_time = gate.getAttribute("open-time")
+
+        start_time = int(start_time.split(':')[0])*3600 + int(start_time.split(':')[1])*60
+        
+        #create a dictionary of names and a list of longitudes and latitudes as the waypoints co-ordinates are stored separate to turnpoint details
+        coords = defaultdict(list)
+        
+        for point in wpoints:    
+            coords[point.getAttribute("name")].append(float(point.getAttribute("longitude")))
+            coords[point.getAttribute("name")].append(float(point.getAttribute("latitude")))
+            
+        # create list of turnpoints    
+        for point in tpoints:
+            lat = coords[point.getAttribute("name")][1]
+            lon = coords[point.getAttribute("name")][0]
+            radius = float(point.getAttribute("radius"))/1000
+            
+            if point.getAttribute("idx") == "0":
+                if point.getAttribute("Exit") == "true":
+                    sort = "start_exit"
+                else:
+                    sort = "start_enter"
+            else:
+                if point == tpoints[-1]:   # if it is the last turnpoint i.e. the goal
+                    if point.getAttribute("type") == "line":
+                        sort = "goal_cylinder"     # to change one line can be processed.
+                    else:
+                        sort = "goal_cylinder"
+                else:
+                    sort = "cylinder"
+            
+                      
+            turnpoint = Turnpoint(lat, lon, radius, sort)
+            turnpoints.append(turnpoint)
+        task = Task(turnpoints, start_time)
+        return task
+    
+    
+    def __init__(self, turnpoints, start_time, end_time=86399):  #finish_time defaults to 23:59
+        self.turnpoints = turnpoints
+        self.start_time = start_time
+        self.end_time = end_time
+        
+        
+        
+    def check_flight(self, flight):
+        """ Checks a flight object against the task. 
+            Args:
+                   flight: a flight object
+            Returns:
+                    a list of rawtimes of when turnpoints were achieved.
+            
+        """
+        turnpoint_times = []   
+    
+        proceed_to_start = False
+        t=0
```

Comment:
I'd call this reached_turnpoints, as this variable stores the number of turnpoints we've visited.